packages/vscode-extension/src/debugging/DebugSession.ts

```diff
@@ -22,134 +22,148 @@ export type DebugSessionDelegate = {
+/**
+ * Helpr function that starts a debug session and returns the session object upon sucesfull start
+ */
+async function startDebugging(
+  folder: vscode.WorkspaceFolder | undefined,
+  nameOrConfiguration: string | vscode.DebugConfiguration,
+  parentSessionOrOptions?: vscode.DebugSession | vscode.DebugSessionOptions
+) {
+  const debugSessionType =
+    typeof nameOrConfiguration === "string" ? nameOrConfiguration : nameOrConfiguration.type;
+  let debugSession: vscode.DebugSession | undefined;
+  let didStartHandler: Disposable | null = debug.onDidStartDebugSession((session) => {
+    if (session.type === debugSessionType) {
+      didStartHandler?.dispose();
+      didStartHandler = null;
+      debugSession = session;
+    }
+  });
+  try {
+    const debugStarted = await debug.startDebugging(
+      folder,
+      nameOrConfiguration,
+      parentSessionOrOptions
+    );
+
+    if (debugStarted) {
+      // NOTE: this is safe, because `debugStarted` means the session started successfully,
+      // and we set the session in the `onDidStartDebugSession` handler
+      assert(debugSession, "Expected debug session to be set");
+      return debugSession;
+    } else {
+      throw new Error("Failed to start debug session");
+    }
+  } finally {
+    didStartHandler?.dispose();
+  }
+}
 
 export class DebugSession implements Disposable {
-  private vscSession: VscDebugSession | undefined;
-  private debugEventsListener: Disposable;
-  private pingTimeout: NodeJS.Timeout | undefined;
-  private pingResolve: ((result: boolean) => void) | undefined;
-  private wasConnectedToCDP: boolean = false;
+  private parentDebugSession: vscode.DebugSession | undefined;
+  private jsDebugSession: vscode.DebugSession | undefined;
+
+  private disposables: Disposable[] = [];
+
   private currentWsTarget: string | undefined;
 
-  constructor(private delegate: DebugSessionDelegate) {
-    this.debugEventsListener = debug.onDidReceiveDebugSessionCustomEvent((event) => {
-      switch (event.event) {
-        case DEBUG_CONSOLE_LOG:
-          this.delegate.onConsoleLog(event);
-          break;
-        case DEBUG_PAUSED:
-          this.delegate.onDebuggerPaused(event);
-          break;
-        case DEBUG_RESUMED:
-          this.delegate.onDebuggerResumed(event);
-          break;
-        case "RNIDE_pong":
-          if (this.pingResolve) {
-            this.pingResolve(true);
-          } else {
-            Logger.warn("[DEBUG SESSION] Received unexpected pong event");
-          }
-          break;
-        case "RNIDE_profilingCPUStarted":
-          this.delegate.onProfilingCPUStarted(event);
-          break;
-        case "RNIDE_profilingCPUStopped":
-          this.delegate.onProfilingCPUStopped(event);
-          break;
-        default:
-          // ignore other events
-          break;
-      }
-    });
+  constructor(private useParentDebugSession: boolean, private delegate: DebugSessionDelegate) {
```

Comment:
NIT: I'd prefer if we passed some `options` object instead of naked booleans as arguments, it improves the readability of the code that calls this.
Currently:
```ts
const session = new DebugSession(true, ...);
```
With options:
```ts
const session = new DebugSession(..., { useParentDebugSession: true });
```


```suggestion
  constructor(private delegate: DebugSessionDelegate, private sessionOptions: {
  useParentDebugSession: boolean
  }) {
```

Reply:
I changed this already